### ISSUE TITLE
Using Ledger wallet: unable to start Web3 call after signing Solana message

### ISSUE THREAD

**Issue by srirambv** [brave/brave-talk#1090]
### Description
Unable to start Web3 call after signing Solana message

### Steps to reproduce

1. Initiate a Web3 call
2. Select Solana
3. Select Ledger Solana account to connect 
4. Click on Start a Web3 Call
5. Sign the message
6. Shows `Error signing message` and doesn't start the call

### Actual Result

https://github.com/brave/brave-talk/assets/17010094/81ee8936-ab56-4210-beee-8b77d92eed75

### Expected Result
Able to start Web3 call without issues

### Additional Information
cc: @mrose17 @HEagle18

**Comment by bbondy** [comment]
Does this reproduce with MetaMask when using Ledger too?

**Comment by bbondy** [comment]
Sorry I meant Phantom

**Comment by srirambv** [comment]
Haven't checked with Ledger+Phantom. Will check and update in a comment

**Comment by srirambv** [comment]
Same issue with Phantom as well but it provides more info as to why its failing rather than just saying `Cannont read properties of undefined`
![image](https://github.com/brave/brave-talk/assets/17010094/e41403bb-45bc-42cf-be71-f3d883e75557)

**Comment by mrose17** [comment]
@srirambv - thanks for checking. at least we know we are "ledger compatible with phantom".

all - i don't know what the root cause is here, i'll do a little research. i'll make sure our FAQ notes this limitation.

**Comment by mrose17** [comment]
@srirambv - can you verify whether "blind signing" is enabled on the Solana Ledger App? if enabled, does the issue occur for either the Brave or Phantom wallets?

**Comment by srirambv** [comment]
@mrose17 the problem is when trying to join the Solana call on a different browser (Brave/Edge with Phantom) it doesn't recognize the Solana account. It automatically selects Ethereum accounts so no way to get it to sign

**Comment by mrose17** [comment]
i think we are talking about two different things.

the video in the original report above shows someone trying to create a call using a SOL address on a ledger wallet and getting an error. can you reproduce that error with "blind signing enabled?

**Comment by mrose17** [comment]
> @mrose17 the problem is when trying to join the Solana call on a different browser (Brave/Edge with Phantom) it doesn't recognize the Solana account. It automatically selects Ethereum accounts so no way to get it to perform Solana sign

@hadiamjad - can you confirm that when someone goes to join a call that was created with SOL NFT token gating, that the initial error response they get indicates two things: "this is a web3 call" and "use a SOL wallet"? it seems that what is reported here is that the second indication isn't sent or is being ignored...

**Comment by srirambv** [comment]
> the video in the original report above shows someone trying to create a call using a SOL address on a ledger wallet and getting an error. can you reproduce that error with "blind signing enabled?

Yes. even with "blind signing"  enabled it still shows the same error message

**Comment by mrose17** [comment]
to confirm:
- this is for someone creating a room (not joining)
- they are using a ledger wallet in which "blind signing" is enabled on the SOL application on the ledger wallet
- this fails for both the Brave and Phantom wallets

**Comment by srirambv** [comment]
>this is for someone creating a room (not joining)
Yes. Trying to create a room with Solana Ledger fails 

> they are using a ledger wallet in which "blind signing" is enabled on the SOL application on the ledger wallet
Yes. With blind signing enabled, creating a room it shows the same error as shown in https://github.com/brave/brave-talk/issues/1090#issuecomment-1645338903

> this fails for both the Brave and Phantom wallets
Yes

**Comment by mrose17** [comment]
@hadiamjad - TL;DR: the ledger hardware wallet doesn't do the kind of signing we want, so this issue is outside of scope for now; however, please look into https://github.com/brave/brave-talk/issues/1090#issuecomment-1645650896 and, if we're not doing this now, please create an issue and then make this change. thanks!

**Comment by t98rex** [comment]
hello is there an solution to this problem? of-late i'm facing this. 
blind signing is enalbled but the signing transaction never makes it to my ledger.

**Comment by mrose17** [comment]
Hello! I am in the process of reviewing Brave Talk issues and this one is very interesting...

**Comment by mrose17** [comment]
@srirambv - may i impose you to see if this still reproduces?

**Comment by srirambv** [comment]
This still seems to be an issue with Ledger accounts

https://github.com/brave/brave-talk/assets/17010094/8e0260af-622f-4c4f-8d95-b97af85689d3

**Comment by mrose17** [comment]
OBE
